### PR TITLE
chore: allow user to view memos in different modes

### PR DESCRIPTION
**Pull request by MehadND**
### This PR addresses the issue around memos that have a lot of content/length. 
Inspired by [this issue](https://github.com/usememos/memos/issues/2863)

The PR offers a way for users to set a mode for their memos in preference settings. 

The two options available to users are `FULL` and `COMPACT`.

**Full - Displays all of memo and its content.
Compact - Displays a fraction of memo and hides the rest.**

In Compact mode, users are given option in dropdown menu to expand/shrink current memo.

_See video for reference._

https://github.com/usememos/memos/assets/34424878/5adb8f09-7383-4716-b5ef-c5db3053c5c7

### CODE REVIEW

**Review by MohamadMirzaeiDev — COMMENTED** [review]
this is great .

What algorithm did you use to view in different modes?

Is this related to the core of the program ?

**Comment by MehadND** [comment]
> this is great .
> 
> 
> 
> What algorithm did you use to view in different modes?
> 
> 
> 
> Is this related to the core of the program ?

Basically I created a string variable in user settings. Most of the implementation was similar to the visibility setting.

Hopefully that answers your questions.

**Comment by MehadND** [comment]
Hey @boojack can you review this PR? and can you please let me know if it can be merged with the current code or would I have to pass the teats?

p.s. i tried fixing the lint errors but couldn't

**Comment by lincolnthalles** [comment]
> Hey @boojack can you review this PR? and can you please let me know if it can be merged with the current code or would I have to pass the teats?
> 
> p.s. i tried fixing the lint errors but couldn't

You're almost there.

You must run `buf generate` inside `/proto`, or `pnpm pnpm postinstall` from `/web` and commit the generated files.
image

Here's a git patch after running `buf generate` in this PR.
[proto-2863.patch](https://github.com/usememos/memos/files/14172229/proto-2863.patch)

After that, `golangci-lint run` will still show some warnings about `unchecked-type-assertion`, but they are not related to your code and it's safe to ignore.

**Comment by MehadND** [comment]
> > Hey @boojack can you review this PR? and can you please let me know if it can be merged with the current code or would I have to pass the teats?
> 
> > 
> 
> > p.s. i tried fixing the lint errors but couldn't
> 
> 
> 
> You're almost there.
> 
> 
> 
> You must run `buf generate` inside `./proto`, or `pnpm pnpm postinstall` from `./web` and commit the generated files.
> 
> image
> 
> 
> 
> Here's a git patch after running `buf generate` in this PR.
> 
> [proto-2863.patch](https://github.com/usememos/memos/files/14172229/proto-2863.patch)
> 
> 
> 
> After that, `golangci-lint run` will still show some warnings about `unchecked-type-assertion`, but they are not related to your code and it's safe to ignore.
> 
> 

Ah ok that really helps a lot, I thought generally I should ignore pushing `.pb.go` files. My bad should have researched and understood the whole codebase.

Thank you for helping.

File: proto/store/user_setting.proto [hunk 1]
```diff
@@ -15,6 +23,7 @@ message UserSetting {
     string appearance = 5;
     string memo_visibility = 6;
     string telegram_user_id = 7;
+    string memo_mode = 8;
```
Comment (boojack) [inline proto/store/user_setting.proto:26]: How about `bool compact_view = 8`?

Comment (MehadND) [reply proto/store/user_setting.proto:26]: Yes it's a great suggestion. I will work on it and push the changes soon. Thank you.

**Review by boojack — CHANGES_REQUESTED** [review]
Sorry for the late review. It's better to split this up into a few PRs, otherwise it's harder to review after too many changes.

1. `.proto` changes with buf generate;
2. Backend changes: services and stores;
3. Frontend changes;

**Pull request closed without merging**